### PR TITLE
Warn when closing magic tag is missed

### LINKED ISSUE #675: Misleading error message if {{#let ...}} is used without closing tag
I was trying to pass variables between components without cluttering the vm:
```html
{{#let bob=undefined}}
<my-component hasItem:to="bob" />

<div class="main-body">
  <other-component wantsItem:from="bob" />
</div>
```
I could not figure out what I was doing wrong partially because this was the error message:
image
This is the same error you'd get without using let if `bob` wasn't on the VM.

The actual problem is that the `let` I'm using is for defining it within its own block and has no close tag so the let var doesn't get added to the scope. That should maybe cause its own error?

It works fine if I remove the `#` or close the block, of course, but as someone new to the `let` expressions I didn't realize it until some time later what I did wrong. I just put `bob` on the VM and avoided `let` at the time.

Thank you for your time,
🍻

### PR DESCRIPTION
**Pull request by cherifGsoul**
This adds warning when a closing magic tag is missed for example the following triggers a warning:
`{{# let foo="bar"}} {{foo}}`

### The changes:
The warning is added to [done](https://github.com/canjs/can-stache/blob/83191d3c312845acc70f103d167ef378625284ca/can-stache.js#L479) function

Fixes #675

### CODE REVIEW

File: can-stache.js [hunk 1]
```diff
@@ -476,7 +476,24 @@ function stache (filename, template) {
 				comment: text
 			});
 		},
-		done: function(lineNo){}
+		done: function(lineNo){
+			//!steal-remove-start
+			// warn if closing magic tag is missed #675
+			var last = state.sectionElementStack[state.sectionElementStack.length - 1];
+			if (process.env.NODE_ENV !== 'production') {
+				if(section instanceof HTMLSectionBuilder) {
```
Comment (bmomberger-bitovi) [inline can-stache.js]: I don't think you need this line here.  The block you copied this from (probabaly `makeRendererAndUpdateSection`) was in a function that was passed `section` as a parameter.  In this function you're just referencing the top level `section`, which is always an `HTMLSectionBuilder`.

Comment (cherifGsoul) [reply can-stache.js]: Yeah, I added this check in the 2nd commit the first one was good.

Comment (cherifGsoul) [reply can-stache.js]: is fixed now

File: can-stache.js [hunk 2]
```diff
@@ -476,7 +476,22 @@ function stache (filename, template) {
 				comment: text
 			});
 		},
-		done: function(lineNo){}
+		done: function(lineNo){
+			//!steal-remove-start
+			// warn if closing magic tag is missed #675
+			var last = state.sectionElementStack[state.sectionElementStack.length - 1];
```
Comment (justinbmeyer) [inline can-stache.js]: move this inside

**Comment by cherifGsoul** [comment]
I close this for https://github.com/canjs/can-stache/pull/692

**Pull request closed without merging**